### PR TITLE
[MERGE] partner_autocomplete_address_extended: remove broken module and ensure base_address_extended effectively works

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
PURPOSE

Remove partner_autocomplete_address_extended and ensure base_address_extended
effectively works.

SPECIFICATIONS

Commits a6e1eb9 and 8a1815b added among a lot of other things
a bridge module between ``partner_autocomplete`` and ``base_address_extended``.
It is used only to redefine the ``_split_street_with_params`` method from
``base_address_extended``. This method is used to find street name, number and
number2 from an address, given a format coming from the country.

However the override completely fucks up the original method purpose and uses
an hardcoded regex coming out of the blue. Parameters like country format is
not taken into account which is annoying when trying to parse country dependent
data.

Tests from ``base_address_extended`` crash completely when used with the
``partner_autocomplete_address_extended`` implementation.

Considering the original complete specifications from commits
"""
For Name field or M2O, gives a list of companies
Data comes from Odoo IAP Service
"""

Or specs found in the original task ID 1867818 pad
"""
If Extended Address module is installed, the street number should be correctly
splitted Or the complete address is put in street2 if impossible to parse
"""

We think that this implementation is broken by design. Indeed there is no
mention of street2 anywhere in the code, and this implementation ensure street
will never be correctly split. We therefore remove it completely.

In this merge we also

  * clean and improve tests in base_address_extended. Purpose is to make tests
    easier to understand and more data oriented. Tests about company / partner
    address fields are added to ensure coherency;
  * reorganize python code of base_address_extended according to guidelines;
  * rename compute / inverse methods of base_address_extended;
  * remove dead code from base partner model;

See sub commits for more details.

LINKS

Task ID 2158302
PR #42678

### CODE REVIEW

File: addons/base_address_extended/tests/test_street_fields.py [hunk 1]
```diff
@@ -15,44 +16,76 @@ def setUpClass(cls):
...
-            'street_number2': street_number2,
-            'country_id': country_id,
+    def assertStreetVals(self, partner, street_data):
+        for key, val in street_data.items():
+            if isinstance(partner[key], models.BaseModel):
+                self.assertEqual(partner[key].id, val, 'Wrongly formatted street field %s: expected %s, received %s' % (key, val, partner[key]))
+            else:
+                self.assertEqual(partner[key], val, 'Wrongly formatted street field %s: expected %s, received %s' % (key, val, partner[key]))
+
+    def test_partner_create(self):
```
Comment (awa-odoo) [inline addons/base_address_extended/tests/test_street_fields.py]: `""" Will test the compute and inverse methods of street fields when creating partner records. """`

File: addons/base_address_extended/tests/test_street_fields.py [hunk 2]
```diff
@@ -15,44 +16,76 @@ def setUpClass(cls):
...
+        # test street_values -> street (inverse)
+        for partner_values, expected_vals in zip(input_values, expected):
+            partner_values['name'] = 'Test'
+            expected_street = partner_values.pop('street')
+            partner_values.update(expected_vals)
+            partner = self.env['res.partner'].create(partner_values)
+            self.assertEqual(partner.street, expected_street)
+            self.assertStreetVals(partner, partner_values)
+
+    def test_partner_write(self):
```
Comment (awa-odoo) [inline addons/base_address_extended/tests/test_street_fields.py]: `""" Will test the compute and inverse methods of street fields when updating partner records. """`

File: addons/base_address_extended/models/res_company.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        address_fields = super(Company, self)._get_company_address_fields(partner)
+        address_fields.update({
+            'street_name': partner.street_name,
+            'street_number': partner.street_number,
+            'street_number2': partner.street_number2,
+        })
+        return address_fields
+
+    def _inverse_street(self):
+        for company in self:
```
Comment (awa-odoo) [inline addons/base_address_extended/models/res_company.py]: The `_inverse_street` in the base module also writes on the `partner_id.street` field.
We don't call super and we don't write that field anymore, I guess that's intended?

I know you just moved the code, but it seems weird.

File: addons/base_address_extended/models/res_partner.py [hunk 4]
```diff
@@ -9,20 +9,19 @@
...
-    street_number = fields.Char('House', compute='_split_street', help="House Number",
-                                inverse='_set_street', store=True)
-    street_number2 = fields.Char('Door', compute='_split_street', help="Door Number",
-                                 inverse='_set_street', store=True)
+    street_name = fields.Char(
+        'Street Name', compute='_compute_street_data', inverse='_inverse_street_data', store=True)
+    street_number = fields.Char(
+        'House', compute='_compute_street_data', inverse='_inverse_street_data', store=True)
+    street_number2 = fields.Char(
+        'Door', compute='_compute_street_data', inverse='_inverse_street_data', store=True)
```
Comment (awa-odoo) [inline addons/base_address_extended/models/res_partner.py]: Could we also rename the compute and inverse methods of "addons/base_address_extended/models/res_company.py " to the same things for consistency ?

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

File: odoo/addons/base/models/res_company.py [hunk 5]
```diff
@@ -114,15 +113,14 @@ def init(self):
         if hasattr(sup, 'init'):
             sup.init()
 
+    def _get_company_address_fields_matching(self):
+        """ Tools method returning a matching: company field name: partner field name """
+        return dict((field_name, field_name)
```
Comment (awa-odoo) [inline odoo/addons/base/models/res_company.py]: Make it a regular list instead.
Will simplify the code a lot and we don't really need that complexity (yet).

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Pull request closed without merging**